crates/wkg/src/oci.rs

```diff
@@ -0,0 +1,185 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use clap::{Args, Subcommand};
+use docker_credential::DockerCredential;
+use oci_distribution::{
+    client::{ClientConfig, ClientProtocol},
+    secrets::RegistryAuth,
+    Reference,
+};
+use oci_wasm::{WasmClient, WasmConfig};
+
+#[derive(Debug, Args)]
+pub struct Auth {
+    /// The username to use for authentication. If no credentials are provided, wkg will load them
+    /// from a local docker config and credential store and default to anonymous if none are found.
+    #[clap(
+        id = "username",
+        short = 'u',
+        env = "WKG_OCI_USERNAME",
+        requires = "password"
+    )]
+    pub username: Option<String>,
+    /// The password to use for authentication. This is required if username is set
+    #[clap(
+        id = "password",
+        short = 'p',
+        env = "WKG_OCI_PASSWORD",
+        requires = "username"
+    )]
+    pub password: Option<String>,
+}
+
+impl Auth {
+    fn into_auth(self, reference: &Reference) -> anyhow::Result<RegistryAuth> {
+        match (self.username, self.password) {
+            (Some(username), Some(password)) => Ok(RegistryAuth::Basic(username, password)),
+            (None, None) => {
+                let server_url = format!("https://{}", reference.registry());
+                match docker_credential::get_credential(&server_url) {
+                    Ok(DockerCredential::UsernamePassword(username, password)) => {
+                        return Ok(RegistryAuth::Basic(username, password));
+                    }
+                    Ok(DockerCredential::IdentityToken(_)) => {
+                        return Err(anyhow::anyhow!("identity tokens not supported"));
+                    }
+                    Err(err) => {
+                        tracing::debug!("Failed to look up OCI credentials: {err}");
+                    }
+                }
+                Ok(RegistryAuth::Anonymous)
+            }
+            _ => Err(anyhow::anyhow!("Must provide both a username and password")),
+        }
+    }
+}
+
+#[derive(Debug, Args)]
+pub struct Common {
+    /// A comma delimited list of allowed registries to use for http instead of https
+    #[clap(
+        long = "insecure",
+        default_value = "",
+        env = "WKG_OCI_INSECURE",
+        value_delimiter = ','
+    )]
+    pub insecure: Vec<String>,
+}
+
+/// Commands for interacting with OCI registries
+#[derive(Debug, Subcommand)]
+pub enum OciCommands {
+    /// Get a component from an OCI registry and write it to a file.
+    Get(GetArgs),
```

Comment:
You referred to this as "pull" in the PR description. I liked that because it also matches the terminology used by the OCI distribution crate, the Docker CLI, and Oras. Can we call this command "pull"?

Reply:
Good catch. I had kept previous terminology from #33 where I was doing something for both warg and OCI

Reply:
I intentionally avoided calling this `pull`. While `wkg oci pull` is clearly similar to `oras pull`, `wkg get` is intended to do more than just pull a single artifact; it is more analogous to something like `go get` or `npm install`.

Reply:
So these commands are all under `wkg oci`. Does that change anything? From what @calvinrp conveyed to me, we wanted to have those under a separate oci subcommand and then we have `load` (can be renamed to `get`) and `publish` at the top level. Did I misunderstand the intent?

Reply:
Ohh sorry I looked at the diff for this change and assumed it was the existing command. No, `pull` is good here.